civis/ml/tests/test_helper.py

```diff
@@ -22,3 +26,164 @@ def test_list_models():
+    m_client.scripts.list_containers_runs.return_value = [
+        Response({'id': _id}) for _id in run_ids]
+    m_client.scripts.list_containers_runs_outputs.return_value = [
+        Response({'object_id': 117, 'object_type': 'File', 'name': 'fname'}),
+        Response({'object_id': 31, 'object_type': 'Project'}),
+        Response({'object_id': 37, 'object_type': 'JSONValue'}),
+    ]
+    return m_client
+
+
+def test_share_model_users():
+    m_client = _create_share_model_client_mock([11])
+
+    resp = helper._share_model(3, [7, 8], 'write', 'users', client=m_client,
+                               send_shared_email=True)
+    assert resp == 'usershare'
+    m_client.scripts.put_containers_shares_users.assert_called_once_with(
+        3, [7, 8], 'write', send_shared_email=True)
+    m_client.files.put_shares_users.assert_called_once_with(
+        117, [7, 8], 'write', send_shared_email=False)
+    m_client.projects.put_shares_users.assert_called_once_with(
+        31, [7, 8], 'write', send_shared_email=False)
+    m_client.json_values.put_shares_users.assert_called_once_with(
+        37, [7, 8], 'write', send_shared_email=False)
+
+
+def test_share_model_groups():
+    m_client = _create_share_model_client_mock([11])
+
+    resp = helper._share_model(3, [7, 8], 'write', 'groups', client=m_client,
+                               send_shared_email=True)
+    assert resp == 'groupshare'
+    m_client.scripts.put_containers_shares_groups.assert_called_once_with(
+        3, [7, 8], 'write', send_shared_email=True)
+    m_client.files.put_shares_groups.assert_called_once_with(
+        117, [7, 8], 'write', send_shared_email=False)
+    m_client.projects.put_shares_groups.assert_called_once_with(
+        31, [7, 8], 'write', send_shared_email=False)
+    m_client.json_values.put_shares_groups.assert_called_once_with(
+        37, [7, 8], 'write', send_shared_email=False)
+
+
+def test_share_model_tworuns():
+    # Check that we grant permission on run outputs for each run
+    m_client = _create_share_model_client_mock([11, 13])
+
+    helper._share_model(3, [7, 8], 'write', 'users', client=m_client)
+
+    m_client.scripts.put_containers_shares_users.assert_called_once_with(
+        3, [7, 8], 'write')
+
+    assert m_client.files.put_shares_users.call_count == 2
+    assert m_client.projects.put_shares_users.call_count == 2
+    assert m_client.json_values.put_shares_users.call_count == 2
+
+
+def test_share_model_project_permissions():
+    # Grant "write" permission on the internal project when
+    # overall "read" permission is requested.
+    m_client = _create_share_model_client_mock([11])
+
+    helper._share_model(3, [7, 8], 'read', 'groups', client=m_client)
+
+    m_client.projects.put_shares_groups.assert_called_once_with(
+        31, [7, 8], 'write', send_shared_email=False)
+
+
+def test_share_model_debug_log():
+    # Debug logs need "write" permission to be shared
```

Comment:
I think you're right in most cases, a person who only needs read permissions won't need access to the debug logs. But I don't think it would ever be a problem to expose the debug logs, and there are some cases where you might expect read permissions to provide log access. Probably the most common is CS sharing one of us on a client job so that we can examine the logs and debug, but I could also imagine clients sharing jobs to debug internally, before they escalate to CS. My preference is to share all run outputs for consistency, but I'm open to discussion.

Reply:
Okay, this makes the code simpler, so I'm happy to make the change. I'm not thinking of anything sensitive in the logs which wouldn't also be in the other model artifacts which would be shared at read level.